src/integrationtests/java/com/aws/greengrass/integrationtests/logmanager/LogManagerTest.java

```diff
@@ -283,4 +291,60 @@ void GIVEN_system_config_with_small_periodic_interval_WHEN_interval_elapses_THEN
         }
         assertEquals(1, allFiles.size());
     }
-}
+
+    @Test
+    void GIVEN_component_configs_WHEN_individual_configs_are_reset_THEN_correct_default_values_are_used()
+            throws Exception {
+        Path LOG_FILE_DIRECTORY_PATH_DEFAULT = tempRootDir.resolve("logs");
+
+        tempDirectoryPath = Files.createTempDirectory(tempRootDir, "IntegrationTestsTemporaryLogFiles");
+        setupKernel(tempDirectoryPath, "configsDifferentFromDefaults.yaml");
+        TimeUnit.SECONDS.sleep(30);
+
+        // Verify correct reset for periodicUploadIntervalSec
+        assertThat(()-> logManagerService.getPeriodicUpdateIntervalSec(), eventuallyEval(is(10), Duration.ofSeconds(30)));
+        logManagerService.getConfig().find("configuration", "periodicUploadIntervalSec").remove();
+        assertThat(()-> logManagerService.getPeriodicUpdateIntervalSec(),
+                eventuallyEval(is(logManagerService.DEFAULT_PERIODIC_UPDATE_INTERVAL_SEC), Duration.ofSeconds(30)));
+
+        // Verify correct reset for fileNameRegex
+        assertThat(()-> logManagerService.getComponentLogConfigurations().get("UserComponentA").getFileNameRegex().pattern(),
+                eventuallyEval(is("^integTestRandomLogFiles.log\\w*"), Duration.ofSeconds(30)));
+        logManagerService.getConfig().find("configuration", "logsUploaderConfiguration",
+                "componentLogsConfigurationMap", "UserComponentA", "logFileRegex").remove();
```

Comment:
nit: can we reference all these config keys from existing string constants? e.g. "logsUploaderConfiguration" -> LOGS_UPLOADER_CONFIGURATION_TOPIC

Reply:
Good call. Changed.